planner/logical.go

```diff
@@ -163,6 +170,29 @@ func (v *fluxSpecVisitor) visitOperation(o *flux.Operation) error {
 	return nil
 }
 
+func validate(plan *PlanSpec) (*PlanSpec, error) {
+	if len(plan.Results()) > 1 {
+		names := make(map[string]struct{}, len(plan.Results()))
+
+		for _, root := range plan.Results() {
+			spec, ok := root.ProcedureSpec().(YieldProcedureSpec)
+
+			if !ok {
+				return nil, errors.New("query must have explicit yields for multiple result")
+			}
+
+			if name, ok := names[spec.YieldName()]; ok {
+				return nil, fmt.Errorf("found duplicate yield name %q", name)
+			}
+
+			if len(root.Predecessors()) != 1 {
+				return nil, errors.New("yield procedures must have exactly one predecessor")
```

Comment:
Is this unreachable?  I guess since `flux.Spec` is part of the public API, it's possible to start with a graph where nodes have the wrong number of predecessors.  Probably we'll want checks like this for each kind of node.